client/src/get-compiled-template.js

```diff
@@ -32,11 +23,20 @@ module.exports = function(cache){
             }
           });
         }
-        var type = template.type;
-        if (type === 'jade') { type = 'oc-template-jade'; }
-        if (type === 'handlebars') { type = 'oc-template-handlebars'; }
 
-        cb(null, templateEngines[type].getCompiledTemplate(templateText, template.key));
+        var type = template.type;
+        var ocTemplate;
+        try {
+          if (type === 'jade') { type = 'oc-template-jade'; }
+          if (type === 'handlebars') { type = 'oc-template-handlebars'; }
+
+          // dynamically require specific oc-template
+          ocTemplate = require(type);
+        } catch (err) {
+          throw format(settings.templateNotSupported, type);
+        }
+        
+        cb(null, ocTemplate.getCompiledTemplate(templateText, template.key));
```

Comment:
I would go with #2 for the moment so that we don't spread around too many pieces. Once out of 'flux', this could definitely go in its own npm module.